sdk/servicebus/azure-messaging-servicebus/src/test/java/com/azure/messaging/servicebus/ServiceBusSenderClientTest.java

```diff
@@ -224,6 +322,35 @@ void scheduleMessage() {
         Assertions.assertEquals(scheduledEnqueueTime, scheduledEnqueueTimeActual);
     }
 
+    /**
+     * Verifies that scheduling a message will result in calling asyncSender.scheduleMessage() with transaction.
+     */
+    @Test
+    void scheduleMessageWithTransaction() {
+        // Arrange
+        final ServiceBusMessage testData =
+            new ServiceBusMessage(TEST_CONTENTS.getBytes(UTF_8));
+        final Instant scheduledEnqueueTime = Instant.now();
+        final long sequenceNumber = 1;
+
+        when(asyncSender.scheduleMessage(testData, scheduledEnqueueTime, transactionContext)).thenReturn(Mono.just(sequenceNumber));
+
+        // Act
+        sender.scheduleMessage(testData, scheduledEnqueueTime, transactionContext);
+
+        // Assert
+        verify(asyncSender, times(1)).scheduleMessage(testData, scheduledEnqueueTime, transactionContext);
+        verify(asyncSender).scheduleMessage(singleMessageCaptor.capture(), scheduleMessageCaptor.capture(), transactionCaptor.capture());
```

Comment:
Do you need to capture it? The 336 already returns only when the parameters are equal.

Reply:
Same with ones above.